### PR TITLE
refactor: extract font-readiness from PagedEditor into core controller

### PR DESCRIPTION
**Pull request by jan-kubica**
**First slice of the God-module extraction** (audit's #1 structural item): carve orchestration out of `packages/react/src/paged-editor/PagedEditor.tsx` (~6,150 lines) into the framework-neutral core controller, leaving the React file as a lifecycle+DOM shell.

## What moved
`collectInitialLayoutFontFaces`, `collectInitialLayoutFontFamilies`, `documentFontsAreLoaded`, `getDocumentFontSet`, `waitForInitialLayoutFonts` (+ their private helpers/constants) → **`@stll/folio-core/controller/fontReadiness`**. This logic decides which font faces a document needs (from its model + ProseMirror content), gates the first layout on them loading, and probes the browser font set.

## Why this slice first
- **Already framework-agnostic** — imported only React-free core modules; nothing React/JSX touched it, it just lived in the `.tsx`.
- **Behavior-preserving by construction** — the existing font-collection unit test (which exercised these functions directly, no editor view) moves to core alongside the code and still passes. Call sites in PagedEditor are byte-identical; only the import source changed.
- **Removes a latent dup** — `documentFontsAreLoaded` is duplicated in the Vue adapter and is already an injected dependency of the core layout pipeline; a fast-follow can point Vue at this same module.
- **Not the hot path** — font collection runs once per load / on `loadingdone`, never in the pagination inner loop.

## Boundary safety
`document.fonts` / `window` are only touched at call time (same pattern as the existing `browserClock`), so the module stays importable in non-browser hosts. `react-free-core`, `layer-boundaries`, and `model-purity` arch tests all stay green.

## Verification
- Moved test passes (6 cases incl. 2 new: SSR/headless guard + null-model default floor)
- `typecheck` (all 7 packages), `lint`, `format:check` clean
- Arch tests green; `layoutPipeline` test (consumes the `documentFontsAreLoaded` seam) green
- Interaction behavior (font-ready re-layout) covered by the CI `interactions` job
- Changeset: core + react patch (internal refactor, no public API change)

Follow-ups (separate PRs): point the Vue adapter's `documentFontsAreLoaded` at this module; then the next slices (`buildLayoutInputSignature`, pointer pipeline, HF lifecycle).

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved initial document layout reliability by ensuring required fonts are detected and loaded before rendering.
  * Added support for server-side and headless environments where browser font APIs are unavailable.
  * Ensured the default layout font is included when document content is missing.

* **Refactor**
  * Centralized font-readiness handling for consistent behavior across supported integrations.

### CODE REVIEW

File: packages/core/src/controller/fontReadiness.ts [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+  );
+  addLayoutFontFamilyFace(
+    faces,
+    documentModel?.package.theme?.fontScheme?.minorFont?.latin,
+    REGULAR_LAYOUT_FONT_DESCRIPTOR,
+  );
+  addTextFormattingFontFaces(faces, documentModel?.package.styles?.docDefaults?.rPr);
+  for (const style of documentModel?.package.styles?.styles ?? []) {
+    addTextFormattingFontFaces(faces, style.rPr);
+  }
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/core/src/controller/fontReadiness.ts:117]: Pushing back: `Document.package` is required by the canonical `@stll/docx-core/model` type. Accepting a partial object here would mask an invalid boundary value and weaken the inference chain; tests that need partial data should construct a valid `Document`. CC on behalf of @jan-kubica

File: packages/core/src/controller/fontReadiness.ts [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+  const loadFonts = Promise.allSettled(loadChecks.map((check) => fontSet.load(check)))
+    .then(() => fontSet.ready)
+    .then(() => true);
+  return Promise.race([
+    loadFonts,
+    new Promise<boolean>((resolve) => {
+      window.setTimeout(() => resolve(false), INITIAL_LAYOUT_FONT_TIMEOUT_MS);
+    }),
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/core/src/controller/fontReadiness.ts:82]: Agreed and implemented in 8ddd825: the timeout now uses `globalThis.setTimeout`, so it remains safe outside a Window environment. CC on behalf of @jan-kubica
